### PR TITLE
TS Type check - strictNullChecks

### PR DESCRIPTION
**Pull request by hizir-bc**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: clients/cobol-lsp-vscode-extension/src/services/Settings.ts [hunk 1]
```diff
@@ -221,7 +222,9 @@ export class SettingsService {
    * @returns lsp port number
    */
   public static getLspPort(): number | undefined {
-    return +vscode.workspace.getConfiguration().get(SERVER_PORT);
+    //ATTENTION WAS THIS INTENTIONAL?
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/Settings.ts]: yes, it is the way to parse a string into a number

File: clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts [hunk 2]
```diff
@@ -455,18 +459,15 @@ export class CopybookDownloadService implements vscode.Disposable {
   }
 
   public async start() {
-    let startTime: number | null = null;
+    const startTime = Date.now();
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]: Are you sure if this equivalent change?
```
await this.queue.pop();
```
looks like blocking call

File: clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts [hunk 3]
```diff
@@ -527,7 +526,8 @@ export class CopybookDownloadService implements vscode.Disposable {
   ) {
     const toDownload: CopybookProfile[] = [element];
     while (this.queue.length > 0) {
-      toDownload.push(await this.queue.pop());
+      const downloadItem = await this.queue.pop();
+      if (downloadItem) toDownload.push(downloadItem);
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]: We usually use the {} event for one-line branches.

File: clients/cobol-lsp-vscode-extension/src/services/reporter/model/TelemetryEvent.ts [hunk 4]
```diff
@@ -12,59 +12,15 @@
...
-    return this._measurements;
-  }
+export interface TelemetryEventMeasurements {
+  readonly [key: string]: number;
+}
 
-  set measurements(value: Map<string, number>) {
-    this._measurements = value;
-  }
+export interface TelemetryEvent {
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/reporter/model/TelemetryEvent.ts:19]: 👍 
Is there a reason why you prefer interface to type?

File: clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts (hunk 3 above)
Comment (hizir-bc) [reply clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]: done 👍

File: clients/cobol-lsp-vscode-extension/src/services/Settings.ts (hunk 1 above)
Comment (hizir-bc) [reply clients/cobol-lsp-vscode-extension/src/services/Settings.ts]: improved 👍

File: clients/cobol-lsp-vscode-extension/src/services/reporter/model/TelemetryEvent.ts (hunk 4 above)
Comment (hizir-bc) [reply clients/cobol-lsp-vscode-extension/src/services/reporter/model/TelemetryEvent.ts:19]: i thought we might need to extend it in the future

File: clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts (hunk 2 above)
Comment (hizir-bc) [reply clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]:
```
if (startTime === null) {
      startTime = Date.now();
}
```

This if block used to check and assign the startTime, however it was null with no doubt, i thought removing the if and assigning it in the beginning would make sense. Am I missing something?

Comment (ishche) [reply clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]: await this.queue.pop(); can take some time(), that will mess up time measurement.
clients/cobol-lsp-vscode-extension/src/services/copybook/DownloadQueue.ts:59 and clients/cobol-lsp-vscode-extension/src/services/copybook/DownloadQueue.ts:83 if I reed it correctly

Comment (hizir-bc) [reply clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts]: valid point, i am reverting this change back, thanks for this good catch!

**Review by ishche — APPROVED** [review]

**Pull request merged**